common/src/main/scala/org/neo4j/spark/util/Validations.scala

```diff
@@ -45,6 +45,99 @@ case class ValidateSchemaOptions(neo4jOptions: Neo4jOptions, schema: StructType)
   }
 }
 
+case class ValidateSchemaMetadataWrite(neo4jOptions: Neo4jOptions, saveMode: SaveMode) extends Validation {
+  override def validate(): Unit = {
+    val schemaMetadata = neo4jOptions.schemaMetadata
+    if (schemaMetadata.optimizationType != OptimizationType.NONE
+      && (schemaMetadata.optimization.nodeUnique
+      || schemaMetadata.optimization.nodeKey
+      || schemaMetadata.optimization.relationshipUnique
+      || schemaMetadata.optimization.relationshipKey
+      || schemaMetadata.optimization.`type`)) {
+      throw new IllegalArgumentException(
+        s"""You cannot combine `${Neo4jOptions.SCHEMA_OPTIMIZATION_TYPE}` with:
+           |- `${Neo4jOptions.SCHEMA_OPTIMIZATION_NODE_UNIQUE}`
+           |- `${Neo4jOptions.SCHEMA_OPTIMIZATION_NODE_KEY}`
+           |- `${Neo4jOptions.SCHEMA_OPTIMIZATION_RELATIONSHIP_UNIQUE}`
+           |- `${Neo4jOptions.SCHEMA_OPTIMIZATION_RELATIONSHIP_KEY}`
+           |- `${Neo4jOptions.SCHEMA_TYPE_CONSTRAINT}`
+      """)
+    }
+    if (schemaMetadata.optimization.nodeUnique && schemaMetadata.optimization.nodeKey) {
+      throw new IllegalArgumentException(
```

Comment:
this could be a warning instead (and then ignore the unique constraint)
but fine by me

Reply:
I usually prefer a fail-fast approach as it could lead to unexpected situations. So I would keep it as is